x/evm/keeper/state_transition.go

```diff
@@ -191,6 +191,7 @@ func (k *Keeper) ApplyTransaction(tx *ethtypes.Transaction) (*types.MsgEthereumT
 		return nil, stacktrace.Propagate(err, "failed to apply ethereum core message")
 	}
 
+	k.ctxStack.FlattenCaches()
```

Comment:
if you flatten the cache here, k.RevertToSnapshot(revision) won't work ?

Reply:
Yeah, it's better to change to sth like `CommitToRevision(revision)`, `FlattenCaches` happens to work here is because `revision` is always zero.

Reply:
I changed to more flexible `CommitToRevision`.